dandi/metadata.py

```diff
@@ -250,6 +250,8 @@ def extract_age(metadata):
         try:
             duration = parse_age(metadata["age"])
         except (KeyError, TypeError, ValueError):
+            if "age" in metadata and metadata["age"] is not None:
+                raise
             return ...
```

Comment:
I think such code would be cleaner/easier to digest, or am I missing some other intention here?
```suggestion
        if metadata.get("age") is not None:
            duration = parse_age(metadata["age"])
        else:
            return ...
```